### PR TITLE
feature/sharing

### PR DESCRIPTION
**Pull request by UnsungHero97**
implementing steps 1 and 2 from the sharing diagram

### CODE REVIEW

File: js/src/search/ObjectSharingService.coffee [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  Validators            = require 'kryptnostic.validators'
+
+  log = Logger.get('ObjectSharingService')
+
+  { validateId, validateUuid } = Validators
+
+  class ObjectSharingService
+
+    constructor: ->
+      @engine               = new MockKryptnosticEngine()
```
Comment (makosblade) [inline js/src/search/ObjectSharingService.coffee]: Have you tested this at all with the real KryptnosticEngine?

Comment (UnsungHero97) [inline js/src/search/ObjectSharingService.coffee]: no, not at all :( Ryan made it seem like he had never tested agains the real `KryptnosticEngine`. there's a task to get everything off of `MockKryptnosticEngine` and switch to the real. Peng is working on that partially, and I'll work on it when sharing is done.

Comment (howpenghui) [inline js/src/search/ObjectSharingService.coffee]: Yeah, I am switching real engine in. Things look good now I just have not
pushed. Will work on it when I am awake in the morning.

Comment (makosblade) [inline js/src/search/ObjectSharingService.coffee]: Nice! Thanks!

**Comment by makosblade** [comment]
LGTM

**Pull request merged**